### ISSUE TITLE
Similiar paths ending with a different static part are not handled properly

### ISSUE THREAD

**Issue by lexor90** [herenow/light-router#2]
Let's have

´/:user_id/:id/photos`and`/:user_id/:id/pages`both responding to the same method, let's say`POST`.

The router will follow the structure only of the first matching path, so when coming to the last staticPath it will find either `photos` or `pages` (according to what has been declared first) and return 404, making them not usable togheter.

Example:

```
app.post('/:user_id/:id/pages', function (req, res) {...});
app.post('/:user_id/:id/photos', function (req, res) { ... });

POST /abc/def/pages -> 200 OK
POST /abc/def/photos -> 404 Not Found
```

I tried to remake it by changing the initial structure to merge toghether static paths under the same dynamic route, but it would not work if :user_id becomes something else.
It should be implemented a runtime logic to try all matching paths, so if the first one fails, but the structure has some other dynamicPath with '3' parts it should also try them before to fail with a not found.

Any idea of how this could be implemented in your logic?

BTW, thanks for this useful library!

**Comment by lexor90** [comment]
As of my understanding it should be implemented inside `router.js` where 

```
table = table.dynamicRoutes[size]
```

if size is greater than 1 instead of simply `return notFound(req, res)` we should be able to reduce the original table by the already parsed element (e.g. `delete table[justLoopedItem]`) and remake the logic with the new table that is reduced by one item (remake the whole `continue_to_next_value` loop). 

This will introduce worse performances if multiple paths are using the same method/structure, but sometimes, especially for RESTful webservices, route design is not a matter of choice and the `/:parent/:child` URL structuring is just better than `/somenamespacetomakeitwork/:parent/:child`.

**Comment by lexor90** [comment]
Hello @herenow any news about this? I could work on a patch if you would give to me some feedback on what should be the best way to achieve this.

Thanks.

**Comment by herenow** [comment]
I'm going to send in a patch today :)

**Comment by herenow** [comment]
I tried to fix this on https://github.com/herenow/light-router/pull/3

But I can't seem to figure an easy way to solve it, without refactoring the whole thing. Unfortunately I won't have time for this.

I don't plan on maintaining this project any further, it was just an experiment at the time.

I urge you to use another router if you need this kind of routes.

Feel free to send PR's though, if you wan't to fix this issue, I'm not sure how to fix it though :)

**Comment by herenow** [comment]
@lexor90 Actually, I gave the problem a rest, and my fix actually works. Although it may not be elegant.

**Comment by lexor90** [comment]
Hi,
thanks. I tried as well several solutions, including a recursive function and async integration, but they seems too hackish as well. It should be replaced how the whole thing is initialized when parsing routes at startup, because that way you'd only have to loop the final staticPath.

Having to loop the whole array of possible solutions is by no way different than recursing into an array-like structure (incrementing the runtime complexity). 

After some extensions over this library I tried out another routing library extracted from expressjs, maybe 'cause of how it works, or due do the fact my services do not have a long list of endpoints but it runs also much faster.

BTW I ran this library in production for 3+ months, and the only things that stressed me out was this issue and the fact that the first url param could be optional (also undefined) that was easily fixed with some check anyway.
Thanks for it.
